components/RichTabs/index.tsx

```diff
@@ -32,8 +34,33 @@ export function RichTabs({
   className,
   ...rest
 }: RadixTabs.TabsProps) {
+  let [tabValue, setTabValue] = useState(rest.defaultValue);
+  const router = useRouter();
+  const {
+    query: { tab },
+  } = router;
+
+  useEffect(() => {
+    setTabValue(
+      Array.isArray(tab)
+        ? tab[0]
+        : typeof tab === "string" && tab.length > 0
+        ? tab
+        : rest.defaultValue
+    );
+  }, [tab]);
+
+  function handleValueChanged(value: string) {
+    setTabValue(value);
+  }
+
```

Comment:
Noice, as a minor nitpick, if there's no initial tab param set, we could add one to the URL according to the active tab 💅🏽 

In this way, the active tab and the URL are always in sync, and if someone shares the URL with for example the `Display user menu` tab active, then whoever goes to that URL sees that tab active too 💭

Reply:
Yeah so in the usage of the RichTab we always (so far) had a `defaultValue` prop set to default to a certain tab.

Regarding keeping the query param <-> tab in sync, yeah that'd be definitely be a nice improvement down the line I think!